packages/core/src/query/getOrderMetadata.ts

```diff
@@ -0,0 +1,48 @@
+import type { QueryOptions } from '@tanstack/query-core'
+import {
+  getOrderMetadata,
+  type GetOrderMetadataErrorType,
+  type GetOrderMetadataReturnType,
+  type GetOrderMetadataParameters,
+} from '../actions/getOrderMetadata.js'
+import type { Config } from '../createConfig.js'
+import type { Evaluate } from '../types/utils.js'
+import { filterQueryOptions, type ScopeKeyParameter } from './utils.js'
+
+export type GetOrderMetadataOptions = Evaluate<
+  GetOrderMetadataParameters & ScopeKeyParameter
+>
+
+export function getOrderMetadataQueryOptions(
+  config: Config,
+  options: GetOrderMetadataOptions,
+) {
+  return {
+    async queryFn({ queryKey }) {
+      const { scopeKey: _, ...parameters } = queryKey[1]
+      const orderMetadata = await getOrderMetadata(config, parameters)
+      return orderMetadata ?? null
+    },
+    queryKey: getOrderMetadataQueryKey({
+      scopeKey: config.state.id,
+      ...options,
+    }),
+  } as const satisfies QueryOptions<
+    GetOrderMetadataQueryFnData,
+    GetOrderMetadataErrorType,
+    GetOrderMetadataData,
+    GetOrderMetadataQueryKey
+  >
+}
+
+export type GetOrderMetadataQueryFnData = Evaluate<GetOrderMetadataReturnType>
+
+export type GetOrderMetadataData = GetOrderMetadataQueryFnData
+
+export function getOrderMetadataQueryKey(options: GetOrderMetadataOptions) {
+  return ['order-metadata', filterQueryOptions(options)] as const
```

Comment:
Looks like this is implemented correctly but can you confirm that the `order_id` is included in the query key in DevTools? This is to prevent conflicting keys overwriting data which is a pain to debug.

Reply:
yup will do so when testing in our internal app, that's the most straightforward path